src/java/org/apache/cassandra/index/sai/disk/vector/VectorMemtableIndex.java

```diff
@@ -205,35 +214,104 @@ public RangeIterator search(QueryContext queryContext, Expression expr, Abstract
     }
 
     @Override
-    public RangeIterator limitToTopResults(QueryContext context, List<PrimaryKey> keys, Expression exp, int limit)
+    public OrderIterator orderBy(QueryContext queryContext, Expression expr, AbstractBounds<PartitionPosition> keyRange, int limit)
+    {
+        assert expr.getOp() == Expression.Op.ANN : "Only ANN is supported for vector search, received " + expr.getOp();
+
+        Bits bits;
+        if (RangeUtil.coversFullRing(keyRange))
+        {
+            // We don't have shadowed keys yet.
```

Comment:
is this comment obsolete?  i haven't gotten to where we handle SKs in this PR but it doesn't look to be this class

Reply:
Yes, it is.